File: LeetCode/704.binary-search.cpp
/*
 * @lc app=leetcode id=704 lang=cpp
 *
 * [704] Binary Search
 */

// @lc code=start
class Solution {
public:
    int search(vector<int>& nums, int target) {

        int s=0, e=nums.size()-1, mid;

        while(s<=e)
        {
            mid=(s+e)/2;
            if(nums[mid]==target)
                return mid;
            else if(nums[mid]<target)
                s=mid+1;
            else
                e=mid-1;
        }

        return -1;


        
    }
};
// @lc code=end

File: LeetCode/704.binary-search.java
/*
 * @lc app=leetcode id=704 lang=java
 *
 * [704] Binary Search
 */

// @lc code=start
class Solution {
    public int search(int[] nums, int x) {

        int n = nums.length;

        int l = 0;
        int r = n - 1;

        while (l <= r) {
            int mid = (l + r) / 2;

            if (nums[mid] == x) {
                return mid;
            } else if (nums[mid] > x) {
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }

        return -1;

    }
}
// @lc code=end
